### PR TITLE
Add component entities

### PR DESCRIPTION
**Pull request by rnestler**

### CODE REVIEW

**Comment by rnestler** [comment]
It's not yet finished, I failed to mark it as WIP when creating it...

**Comment by rnestler** [comment]
@dbrgn Ready for review.

File: entities/component.py [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+            ' {} {} {} {}\n'.format(self.position, self.rotation, self.required, self.suffix)
+        for pin in self.pins:
+            ret += ' {}\n'.format(pin)
+        ret += ')'
+        return ret
+
+
+class Norm(EnumValue):
+    EMPTY = '""'
+    # TODO not sure about those
```
Comment (dbrgn) [inline entities/component.py]: What's the open question here?

Comment (rnestler) [reply entities/component.py]: What the string value of the other possible values is. Would need to check in the file.

Comment (rnestler) [reply entities/component.py]: Fixed.

**Pull request merged**